### PR TITLE
Add --version to the bin/cake console

### PR DESCRIPTION
**Pull request by alysson-azevedo**
Add `--version` parameter CommandList shell.
It prints the current version of CakePHP.

Add `--version` and `version` parameter to ShellDispatcher.
It performs an internal dispatch to CommandList shell.

See #8891.

### CODE REVIEW

File: src/Shell/CommandListShell.php [hunk 1]
```diff
@@ -136,6 +142,9 @@ public function getOptionParser()
         )->addOption('xml', [
             'help' => 'Get the listing as XML.',
             'boolean' => true
+        ])->addOption('version', [
+            'help' => 'Prints the current version of CakePHP.',
```
Comment (ravage84) [inline src/Shell/CommandListShell.php]: current => installed

Comment (dereuromark) [inline src/Shell/CommandListShell.php]: Or "currently installed" :)

**Comment by alysson-azevedo** [comment]
Alright, i'll fix it. Should I also follow the RFC #8397 and remove all those empty() in the [CommandList](https://github.com/alysson-azevedo/cakephp/blob/637c05c821b3c18fc6209fdd5f22ab17326053cc/src/Shell/CommandListShell.php#L45) file?

File: src/Shell/CommandListShell.php [hunk 2]
```diff
@@ -68,6 +69,11 @@ public function main()
             return;
         }
 
+        if (!empty($this->params['version'])) {
```
Comment (ravage84) [inline src/Shell/CommandListShell.php]: If we had a ``param()`` method, we could make this more readable **and** get rid of the use of ``empty()``.

**Comment by dereuromark** [comment]
Since those could be empty I am fine with keeping the empty() here.
Dont we have param()? I thought we did.

**Comment by ravage84** [comment]
@dereuromark  it seems there is
http://api.cakephp.org/3.2/class-Cake.Console.Shell.html#_param

File: src/Shell/CommandListShell.php (hunk 2 above)
Comment (ravage84) [inline src/Shell/CommandListShell.php]: ``if ($this->param('version') === true) {``

**Comment by dereuromark** [comment]
So

   if ($this->param('version') !== null) {}

should work.

**Comment by alysson-azevedo** [comment]
The reason that `if ($this->params['xml'])` works is because `$this->params` always has all the options defined on `getOptionParser()`. 
See the [Shell::runCommand()](https://github.com/alysson-azevedo/cakephp/blob/637c05c821b3c18fc6209fdd5f22ab17326053cc/src/Console/Shell.php#L413) method.

So, since `$this->params['xml']` and `$this->params['version']` are required to the `CommandList` work properly, if someone change `runCommand()` and `$this->params` becomes empty, phpunit needs to warn that the key doesn't exists.

File: src/Shell/CommandListShell.php [hunk 3]
```diff
@@ -41,7 +42,7 @@ class CommandListShell extends Shell
      */
     public function startup()
     {
-        if (empty($this->params['xml'])) {
+        if (empty($this->params['xml']) && empty($this->params['version'])) {
```
Comment (markstory) [inline src/Shell/CommandListShell.php]: This could be `!$this->param('version')`

**Comment by markstory** [comment]
Thanks :+1:

**Pull request merged**